### PR TITLE
feat: add nix docker image

### PR DESCRIPTION
**Pull request by vdovhanych**
This add nix docker image ported from [vdovhanych/docker-nix](https://github.com/vdovhanych/nix-docker)

### CODE REVIEW

File: nix/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+FROM alpine as build
+
+# Enable HTTPS support in wget and set nsswitch.conf to make resolution work within containers
+RUN apk add --no-cache --update openssl git \
+  && rm -rf /var/cache/apk/* \
+  && echo hosts: files dns > /etc/nsswitch.conf
+
+# Download Nix and install it into the system.
+ARG NIX_VERSION=2.5.1
```
Comment (tsusanka) [inline nix/Dockerfile:11]: @vdovhanych should we stich to latest 2.7.0?

File: nix/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+FROM build
+
+ONBUILD ENV \
+    ENV=/etc/profile \
+    USER=root \
+    PATH=/nix/var/nix/profiles/default/bin:/nix/var/nix/profiles/default/sbin:/bin:/sbin:/usr/bin:/usr/sbin \
+    GIT_SSL_CAINFO=/etc/ssl/certs/ca-certificates.crt \
+    NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
+
+RUN --mount=from=verify,src=/etc/nix,target=/etc/nix /bin/true # hack so buildx does not skip the `verify` step
```
Comment (tsusanka) [inline nix/Dockerfile:49]: newline :)

File: nix/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# docker-builder
+Repository for building and pushing docker image with Nix installed on Alpine. 
+
+This repository is adapted from the depreciated [nixos/docker](https://github.com/nixos/docker)
```
Comment (tsusanka) [inline nix/README.md:4]: deprecated

Comment (tsusanka) [reply nix/README.md:4]: Missing dot at the end of a sentence

File: nix/Dockerfile (hunk 1 above)
Comment (vdovhanych) [reply nix/Dockerfile:11]: its already on 2.6.0 which should be the stable release I was messing with it yesterday but directly to master until I iron out the setup here.

File: nix/Dockerfile (hunk 2 above)
Comment (vdovhanych) [reply nix/Dockerfile:49]: fixed

File: nix/README.md (hunk 3 above)
Comment (vdovhanych) [reply nix/README.md:4]: will be fixed in the next push

File: nix/Dockerfile (hunk 1 above)
Comment (tsusanka) [reply nix/Dockerfile:11]: Great 👍

**Pull request merged**